src/components/swap/GasEstimateTooltip.tsx

```diff
@@ -0,0 +1,61 @@
+import { sendAnalyticsEvent } from '@uniswap/analytics'
+import { InterfaceElementName, SwapEventName } from '@uniswap/analytics-events'
+import { Currency, TradeType } from '@uniswap/sdk-core'
+import { LoadingOpacityContainer } from 'components/Loader/styled'
+import { RowFixed } from 'components/Row'
+import { MouseoverTooltip, TooltipSize } from 'components/Tooltip'
+import { InterfaceTrade } from 'state/routing/types'
+import styled from 'styled-components/macro'
+import { ThemedText } from 'theme'
+
+import { ReactComponent as GasIcon } from '../../assets/images/gas-icon.svg'
+import SwapRoute from './SwapRoute'
+
+const StyledGasIcon = styled(GasIcon)`
+  margin-right: 4px;
+  height: 18px;
+  & > * {
+    stroke: ${({ theme }) => theme.textTertiary};
+  }
+`
+
+export default function GasEstimateTooltip({
+  trade,
+  loading,
+  disabled,
+}: {
+  trade: InterfaceTrade<Currency, Currency, TradeType> // dollar amount in active chain's stablecoin
+  loading: boolean
+  disabled?: boolean
+}) {
+  const formattedGasPriceString = trade?.gasUseEstimateUSD
```

Comment:
another place where formatting function might be nice

Reply:
Yeah, we're doing that a couple of times. FWIW Tina is tackling in her follow-up by refactoring this logic.